### PR TITLE
Introduced the skeleton method of versioning

### LINKED ISSUE #263: Use the python skeleton's version determination method
The python3-pip-skeleton uses a much [simpler method](https://github.com/DiamondLightSource/python3-pip-skeleton/blob/main/src/python3_pip_skeleton/__init__.py) of versioning than `event-model` does  currently with `_version.py`. We should replace this method with the skeleton's.

### PR DESCRIPTION
**Pull request by evalott100**
Closes #263 

Changes to using the importlib metadata method of versioning.

### CODE REVIEW

File: event_model/__init__.py [hunk 1]
```diff
@@ -1759,9 +1763,13 @@ def _is_array(checker, instance):
         for name, schema in schemas.items()
     }
 
+if sys.version_info < (3, 8):
+    from importlib_metadata import version  # noqa
+else:
+    from importlib.metadata import version  # noqa
```
Comment (coretl) [inline event_model/__init__.py:1769]: I'd put this in with the other importlib import at the top of the file

Comment (prjemian) [reply event_model/__init__.py:1769]: Both branches call the same import.  Why is there an `if..else` clause?

Comment (prjemian) [reply event_model/__init__.py:1769]: Removed my comment.  The replacement of `_` by `.` is subtle.

File: event_model/__init__.py [hunk 2]
```diff
@@ -22,14 +22,21 @@
...
-from ._version import get_versions
-
 if sys.version_info < (3, 8):
-    from importlib_metadata import metadata
+    from importlib_metadata import metadata, version  # noqa
 else:
-    from importlib.metadata import metadata
+    from importlib.metadata import metadata, version  # noqa
+
+__version__ = version("event-model")
```
Comment (coretl) [inline event_model/__init__.py:30]: I would put the del here too:

```suggestion
__version__ = importlib_version("event-model")
del importlib_version
```

**Review by coretl — APPROVED** [review]
Do you still need the `#noqa`? Apart from that this looks fine.

**Comment by evalott100** [comment]
Seems not, I'll get rid of it and merge

**Pull request merged**